File: 1288_remove_covered_intervals/remove_covered.java
import java.util.Arrays;

class Solution {
    // Time: O(n * log(n))
    // Space: O(1)
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        int res = 0, left = -1, right = -1;
        for (int[] interval : intervals) {
            if (interval[0] > left && interval[1] > right) {
                res++;
                left = interval[0];
            }
            right = Math.max(right, interval[1]);
        }
        return res;
    }
}

File: 1288_remove_covered_intervals/remove_covered.py
from typing import List


def removeCoveredIntervals(intervals: List[List[int]]) -> int:
    """
    Time: O(n * log(n))
    Space: O(1)
    """
    intervals.sort(key=lambda x: (x[0], -x[1]))
    end = res = 0
    for _, y in intervals:
        if y > end:
            res += 1
            end = y
    return res
